fn.go

```diff
@@ -0,0 +1,230 @@
+package secp256k1
+
+/*
+
+// NOTE: Requires the gmp c library.
+#define USE_NUM_GMP
+
+#define HAVE___INT128
+#define USE_SCALAR_4X64
+#define USE_SCALAR_INV_BUILTIN
+
+#include "secp256k1/include/secp256k1.h"
+#include "secp256k1/src/util.h"
+#include "secp256k1/src/num_gmp_impl.h"
+#include "secp256k1/src/scalar.h"
+#include "secp256k1/src/scalar_impl.h"
+#include "secp256k1/src/scalar_4x64_impl.h"
+
+*/
+import "C"
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"unsafe"
+
+	"github.com/renproject/surge"
+)
+
+// Fn represents an element of the field defined by the prime N, where N is the
+// order of the elliptic curve group secp256k1.
+type Fn struct {
+	inner C.secp256k1_scalar
+}
+
+// NewFnFromUint returns a new field element equal to the given unsigned
+// integer.
+func NewFnFromUint(v uint16) Fn {
+	x := Fn{}
+	x.SetUint(v)
+	return x
+}
+
+// RandomFn returns a random field element.
+//
+// Panics: This function will panic if there was an error reading bytes from
+// the random source.
+func RandomFn() Fn {
+	x, err := RandomFnSafe()
+	if err != nil {
+		panic(fmt.Sprintf("could not generate random bytes: %v", err))
+	}
+	return x
+}
+
+// RandomFnSafe returns a random field element or an error.
+func RandomFnSafe() (Fn, error) {
```

Comment:
In Go, unsafe-ness is already an idiomatic concept (see `package unsafe`). Consider renaming to `RandomFnNoPanic`, which will not be confused with unsafe-ness and is more descriptive of the behaviour.